python/lsst/ts/cbp/component.py

```diff
@@ -94,28 +102,55 @@ def __init__(self):
         self.azimuth = None
         self.mask = None
         self.mask_rotation = None
-
         self.focus = None
+
+        self.altitude_target = 0
+        self.azimuth_target = 0
+        self.mask_target = "Mask 1"
+        self.mask_rotation_target = 0
+        self.focus_target = 0
+
         self.host = None
         self.port = None
         self.panic_status = None
         self.auto_park = None
         self.park = None
         self.simulation_mode = 0
-        self.encoder_status = Status
+        self.encoder_status = Status()
+        self.encoder_motion = Motion()
+        self.in_motion = False
+        self.connected = False
         self.generate_mask_info()
         self.log.info("CBP component initialized")
 
     def generate_mask_info(self):
-        """Generates initial mask info.
-        """
+        """Generates initial mask info."""
         mask_dict = {
             f"{i}": types.SimpleNamespace(name=f"Empty {i}", rotation=0, id=i)
             for i in (1, 2, 3, 4, 5, 9)
         }
         mask_dict["9"].name = "Unknown"
         self.masks = types.SimpleNamespace(**mask_dict)
-        self.log.info(f"{self.masks}")
+
+    def get_motion_status(self):
+        self.encoder_motion.AZIMUTH = self.azimuth != self.azimuth_target
```

Comment:
On the other hand, if you are going to report this value as a SAL event, then I suggest you use the event to cache the information. You already have it in the CSC; you'd have to feed it to the model. That avoids needlessly duplicating cached data.

If your CSC has an axesInPosition event that you pass to the model, then whenever you start a move you set:
```
self.evt_axesInPosition.set_put(azimuth=False)
```

If code wants the value that was most recently reported it can read the cache using:
```
azimuth_in_position = self.evt_axesInPosition.data.azimuth
```

And in the code that polls the actuators:
```
azimuth_in_position = ... read azimuth encoder and "is moving" and compute the value, as above
...likewise for the other axes...
self.evt_axesInPosition.set_put(
    azimuth=azimuth_in_position,
    elevation=elevation_in_position,
    ....
)
```
You already should be reporting this, and it avoids a mismatch between your model and what your CSC reports.

Reply:
1. No.
2. Added error tolerance
Improved in_position definition based on suggestions.